server/server.go

```diff
@@ -235,14 +252,13 @@ func (s *Server) Stop() {
 
 // Close destroys the server
 func (s *Server) Close() {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(s.globalConfiguration.LifeCycle.GraceTimeOut))
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
```

Comment:
Why the *magic* time number here instead of using `GraceTimeOut` ?

Reply:
In fact, this timeout is for the stop of goroutines in Providers and Metrics, not for `GraceTimeOut` on entry points.
Moreover it's a technical timeout, there is no reason to configure it.

Reply:
Are we certain that those 10 seconds are always enough for providers to finish?

Reply:
In reality, I reuse the previous DefaultValue of GraceTimeOut. So, if you don't override the GraceTimeout, it will not really change the behavior. Moreover, as I change the os.Exit to panic, I think we will have more feedback if some goroutines never end or needs a bigger timeout (without change the "real" behavior, because os.Exit or panic will only change the "logs".

Reply:
Where is `GraceTimeOut` being used now? I see that the PR removes an existing usage but does not add it anywhere again.

Reply:
`GraceTimeOut` is used when we close entrypoints. https://github.com/containous/traefik/blob/526a04d4c8c0ad042c69c3ec7c48ab373e97627b/server/server.go#L221